metricbeat/module/system/socket_summary/socket_summary.go

```diff
@@ -0,0 +1,88 @@
+package socket_summary
+
+import (
+	"syscall"
+
+	"github.com/elastic/beats/libbeat/common"
+
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/metricbeat/mb"
+
+	"github.com/shirou/gopsutil/net"
+)
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each host defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet("system", "socket_summary", New,
+		mb.WithNamespace("system.socket.summary"),
```

Comment:
In case we would also use `summary` in the socket metricset it would so we shouldn't. We do the same trick in some other metricsets to make it easier for the user to access the metrics. 

Should we create a collision on day by accident, CI should detect multiple definitions of the same field.